virtual_rainforest/models/abiotic/model.py

```diff
@@ -0,0 +1,146 @@
+"""API documentation for the :mod:`abiotic.model` module.
+************************************************** # noqa: D205
+
+The :mod:`abiotic.model` module creates a
+:class:`~virtual_rainforest.abiotic.model.AbioticModel`
+class as a child of the :class:`~virtual_rainforest.core.model.BaseModel` class.
+"""
+
+from __future__ import annotations
+
+from typing import Any
+
+import pint
+from numpy import datetime64, timedelta64
+
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.model import BaseModel, InitialisationError
+
+
+class AbioticModel(BaseModel):
+    """A class describing the abiotic model.
+
+    Describes the specific functions and attributes that the abiotic module should
+    possess. It's very much incomplete at the moment, and only overwrites one function
+    in a pretty basic manner. This is intended as a demonstration of how the class
+    inheritance should be handled for the model classes.
+
+    Args:
+        update_interval: Time to wait between updates of the model state.
+        soil_layers: The number of soil layers to be modelled.
+        canopy_layers: The initial number of canopy layers to be modelled.
+    """
+
+    model_name = "abiotic"
+
+    def __init__(
+        self,
+        update_interval: timedelta64,
+        start_time: datetime64,
+        soil_layers: int,
+        canopy_layers: int,
+        **kwargs: Any,
+    ):
+
+        if soil_layers < 1:
+            to_raise = InitialisationError(
+                "There has to be at least one soil layer in the abiotic model!"
+            )
+            LOGGER.critical(to_raise)
+            raise to_raise
+
+        elif soil_layers != int(soil_layers):
```

Comment:
*Really* minor style point here, but I'd personally use `if` here rather than `elif`, as the function will already have exited if the first `if` statement was true (via the `raise to_raise`). It's fine as is though